awscli/customizations/cloudfront.py

```diff
@@ -50,14 +63,64 @@ def __init__(self):
 
     def add_to_params(self, parameters, value):
         if value is not None:
-            caller_reference = 'cli-%s-%s' % (
-                int(time.time()), random.randint(1, 1000000))
             parameters['InvalidationBatch'] = {
-                "CallerReference": caller_reference,
+                "CallerReference": caller_reference(),
                 "Paths": {"Quantity": len(value), "Items": value},
                 }
 
 
+def _add_origin_domain_name(argument_table, **kwargs):
+    argument_table['distribution-config'].required = False
+    argument_table['origin-domain-name'] = OriginDomainNameArgument()
+
+
+class OriginDomainNameArgument(CustomArgument):
+
+    def __init__(self):
+        doc = (
+            'The domain name for your origin. Note: --origin-domain-name '
+            'and --distribution-config are mututally exclusive.'
+        )
+        super(OriginDomainNameArgument, self).__init__(
+            'origin-domain-name', help_text=doc)
+
+    def add_to_params(self, parameters, value):
+        if value is None:
+            return
+        origin_id = caller_reference(prefix='origin_id')
```

Comment:
Also, this function should probably be renamed.  It's now no longer used for just `caller_preference`.